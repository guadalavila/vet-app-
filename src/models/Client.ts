export type Client = {
    adress: string;
    comment: string;
    dni: string;
    email: string;
    lastName: string;
    name: string;
    phone: string;
    _id: string;
    createdAt: string;
    updatedAt: string;
};
