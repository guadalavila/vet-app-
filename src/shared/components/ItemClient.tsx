import React, { useContext } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import CustomText from './CustomText';
import { Client } from '~models/Client';
import { colors } from '~shared/utils/colors';
import { size } from '~shared/utils/size';
import { ThemeContext } from '~contexts/ThemeContext';

interface IItemClientProps {
    client: Client;
    onPress: () => void;
}

const ItemClient = ({ client, onPress }: IItemClientProps) => {
    const { themeApp } = useContext(ThemeContext);
    return (
        <TouchableOpacity
            testID='ItemClient'
            activeOpacity={0.7}
            style={[styles.container, { backgroundColor: themeApp.colors.backgroundInput }]}
            onPress={onPress}>
            <CustomText style={styles.text} testID='nameAndLastName'>
                {client.name} {client.lastName}
            </CustomText>
        </TouchableOpacity>
    );
};

export default ItemClient;

const styles = StyleSheet.create({
    container: {
        paddingVertical: size.XL,
        paddingHorizontal: size.XL,
        borderBottomColor: colors.light.primary,
        borderBottomWidth: 1,
    },
    text: {
        fontWeight: 'bold',
    },
});
