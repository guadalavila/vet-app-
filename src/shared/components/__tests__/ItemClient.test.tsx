import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import ItemClient from '../ItemClient';
import { Client } from '~models/Client';

describe('ItemClient', () => {
    const mockFn = jest.fn();

    test('should match snapshot', () => {
        const container = render(<ItemClient client={mockClient} onPress={mockFn} />);
        expect(container).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<ItemClient client={mockClient} onPress={mockFn} />);
        expect(screen.getByTestId('ItemClient')).toBeTruthy();
    });
    test('should render texts', () => {
        render(<ItemClient client={mockClient} onPress={mockFn} />);
        expect(screen.getByTestId('nameAndLastName').children).toEqual(['Juan', ' ', 'Pérez']);
    });

    test('should call fn when press component', () => {
        render(<ItemClient client={mockClient} onPress={mockFn} />);
        const component = screen.getByTestId('ItemClient');
        fireEvent.press(component);
        expect(mockFn).toHaveBeenCalled();
    });
});

const mockClient: Client = {
    _id: '123456789',
    vetId: 'vet123',
    dni: '12345678A',
    name: 'Juan',
    lastName: 'Pérez',
    email: 'juan@example.com',
    phone: '123-456-7890',
    address: 'Calle Falsa 123',
    comment: 'Preferred contact method: email',
    createdAt: '2023-08-26T10:00:00Z',
    updatedAt: '2023-08-26T14:30:00Z',
};
