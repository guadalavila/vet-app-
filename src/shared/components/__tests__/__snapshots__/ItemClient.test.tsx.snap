// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemClient should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ECEDF0",
      "borderBottomColor": "#A765C1",
      "borderBottomWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
  testID="ItemClient"
>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
    testID="nameAndLastName"
  >
    Juan
     
    PÃ©rez
  </Text>
</View>
`;
